discovery/discovery.go

```diff
@@ -54,6 +54,9 @@ type DiscovererOptions struct {
 	// Extra HTTP client options to expose to Discoverers. This field may be
 	// ignored; Discoverer implementations must opt-in to reading it.
 	HTTPClientOptions []config.HTTPClientOption
+
+	// ConfigName is the config key (setName/job name) used for metrics labeling.
```

Comment:
It makes me uncomfortable that this thing has four names.  I think it's called "setName" in the scrape code, and "job" in metrics labels. Could we use one of those for the name here?
Why is "key" appropriate?

Reply:
Good callout, I standardized on SetName here: 00f7ed868433b995d06e31c6f04be9fd0afed82d

"key" was probably not the right term to use either so I removed that.